python/vineyard/core/client.py

```diff
@@ -91,21 +100,47 @@ def __init__(
             socket = os.getenv('VINEYARD_IPC_SOCKET', None)
         if not endpoint and not (host and port):
             endpoint = os.getenv('VINEYARD_RPC_ENDPOINT', None)
-        if endpoint:
-            if not isinstance(endpoint, (tuple, list)):
-                endpoint = endpoint.split(':')
```

Comment:
It is by design to support `connect("....vineyard.sock")` and `connect("127.x.x.x:9600")` at the same time.